### PR TITLE
object : add ssl ref in cephobjectstore user secret

### PR DESCRIPTION
**Pull request by thotz**
There is no reference for ssl in cephobjectstore Secret, so users won't have much idea why tls secret need to used. Hence give reference object stores tls secret ref in the Secret.

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: pkg/operator/ceph/object/user/controller.go [hunk 1]
```diff
@@ -472,13 +478,16 @@ func generateStatusInfo(u *cephv1.CephObjectStoreUser) map[string]string {
...
-func (r *ReconcileObjectStoreUser) generateCephUserSecret(u *cephv1.CephObjectStoreUser) *corev1.Secret {
+func (r *ReconcileObjectStoreUser) generateCephUserSecret(u *cephv1.CephObjectStoreUser, tlsSecretName string) *corev1.Secret {
 	// Store the keys in a secret
 	secrets := map[string]string{
 		"AccessKey": r.userConfig.Keys[0].AccessKey,
 		"SecretKey": r.userConfig.Keys[0].SecretKey,
 		"Endpoint":  r.objContext.Endpoint,
 	}
+	if tlsSecretName != "" {
+		secrets["SSLCertRef"] = tlsSecretName
```
Comment (galexrt) [inline pkg/operator/ceph/object/user/controller.go:489]: What format will the secret name be in? The secret is part of the Rook cluster namespace, so just the secret name might not be useful here. What about including the namespace in `NAMESPACE/SECRET_NAME` format?

Due to RBAC, most applications service accounts wouldn't be able to access the secret in the Rook cluster namespace anyway. Doesn't the endpoint/ URL change accordingly if SSL is enabled or not with `http`/ `https`?

**Comment by parth-gr** [comment]
Does this will solve the fqdn with SSl problem?

File: pkg/operator/ceph/object/user/controller.go [hunk 2]
```diff
@@ -252,7 +252,13 @@ func (r *ReconcileObjectStoreUser) reconcile(request reconcile.Request) (reconci
 	}
```
Comment (parth-gr) [inline pkg/operator/ceph/object/user/controller.go:252]: Commitment complains, Change the commit to 

object: add ssl ref in cephobjectstore user secret

**Review by parth-gr — CHANGES_REQUESTED** [review]

File: pkg/operator/ceph/object/user/controller.go (hunk 1 above)
Comment (thotz) [reply pkg/operator/ceph/object/user/controller.go:489]: We can pass the entire cert as well, what do you think is optimal for sharing the cert directly or pass k8s secret with namespace?

If the Secure port is enable in CephObjectStore CRD, then the endpoint passed in CephObjectStoreUser secret will be https

**Comment by thotz** [comment]
> Does this will solve the fqdn with SSl problem?

Nope, it is just reference for the user to get tls cert if the endpoint is https.

File: pkg/operator/ceph/object/user/controller.go (hunk 1 above)
Comment (thotz) [reply pkg/operator/ceph/object/user/controller.go:489]: @travisn Yes, that's correct. Currently `CephObjectStoreUser` and its secret can only be created where `CephObjectStore` resides where the tls secret is also present

**Review by travisn — APPROVED** [review]

**Review by parth-gr — APPROVED** [review]
Looks Good

**Pull request merged**